src/components/pie/index.js

```diff
@@ -90,14 +94,87 @@ export class PieChartComponent extends events.EventEmitter {
           }
         });
 
-        that.emit('loaded', that, data);
         that.emit('ready', that, data, null);
       })
       .catch((error) => {
         that.emit('error', that, error);
         that.emit('ready', that, null, error);
       });
   }
+
+  _getData(endpoint, cube, params, maxSlices) {
+    var that = this;
+    that.emit('loading', that);
+
+    var api = this.getApiInstance();
+    return api.aggregate(endpoint, cube, params)
+      .then((data) => that._groupSlicesIfMoreThan(data, maxSlices))
+      .then((data) => {
+        that.emit('loaded', that, data);
+        return data;
+      })
+      .catch((error) => {
+        that.emit('error', that, error);
+        that.emit('ready', that, null, error);
+      });
+  }
+
+  _groupSlicesIfMoreThan(data, maxSlices) {
+    if (maxSlices <= 0) {
+      throw RangeError(`Can't create a pie chart with less than 1 slice (asked for ${maxSlices} slices)`);
+    }
+    if (!_.isArray(data.cells) || maxSlices === undefined || data.cells.length <= maxSlices) {
+      return data;
+    }
+
+    const individualCells = data.cells.slice(0, maxSlices - 1);
+    const othersCell = {
+      // Set all dimension's keys that end in "Value" to "Other"
+      dimensions: data.cells[0].dimensions.map((dimension) => {
+        const theDimension = {};
+        Object.keys(dimension).forEach((dimensionKey) => {
+          const value = (dimensionKey.endsWith('Value')) ? this.i18n('others') : dimension[dimensionKey];
+          theDimension[dimensionKey] = value;
+        });
+        return theDimension;
+      }),
+
+      // Set measures as the total values minus the other cell's values
+      measures: Object.keys(data.summary).reduce((measures, measureKey) => {
```

Comment:
While reviewing the code, I figured out a way to simplify this code. Although it still uses a functional style, it's much more readable IMO. What do you think?